src/state.py

```diff
@@ -185,6 +186,18 @@ def _get_agent_meta_map_from_relation(
     return unit_metadata_mapping
 
 
+def _is_auth_proxy_integrated(relation: typing.Optional[ops.Relation]) -> bool:
+    """Check if there is an auth proxy integration..
+
+    Args:
+        relation: The auth-proxy relation.
+
+    Returns:
+        True if an integration for atuh proxy exists.
+    """
+    return bool(relation and relation.app and relation.data[relation.app])
```

Comment:
I think the data for the auth proxy relationship is configured on the requirer side and there's no data on the provider side https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/auth_proxy/v0/README.md. So, to verify the existence of data, `relation.data[self.app]` should be used rather than `relation.data[relation.app]`.

Reply:
I don't need to check if the relation data is present, since existence of the relation suffices so add authentication. changing it. Thanks for spotting this!